includes/express-checkout/class-wc-payments-express-checkout-button-helper.php

```diff
@@ -329,6 +329,26 @@ public function get_product() {
 		return null;
 	}
 
+	/**
+	 * Used to get the order in admin edit page.
+	 *
+	 * @return WC_Order|WC_Order_Refund|bool
+	 */
+	public function get_current_order() {
+		global $theorder;
+		global $post;
+
+		if ( is_object( $theorder ) ) {
```

Comment:
Is there any value in checking the type of `$theorder` or `wc_get_order` here using something like `instanceof`? Or perhaps checking on the result of calling `get_current_order`? I wouldn't consider this a blocker as we have the `is_object` check in all these instances which would filter out the `bool` returned.

Reply:
Oh I should have mentioned in the PR description, but this method in particular, was copied from the PRB integration ([src](https://github.com/Automattic/woocommerce-payments/blob/78fba818fc3462ce4f475041a03cff0256ce2cad/includes/class-wc-payments-payment-request-button-handler.php#L687-L700)). It was introduced more recently in #8150 but it has been validated and apparently has been working properly since then.

Based on that, we could keep it as is, unless you strongly prefer adding an extra check.

Reply:
Nope, that's totally acceptable to me! Thanks for the context on it!